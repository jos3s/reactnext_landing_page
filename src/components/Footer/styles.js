import styled from 'styled-components';

export const Container = styled.footer`
  text-align: center;
  a {
    color: inherit;
    text-decoration: none;
    font-size: ${({ theme }) => theme.font.sizes.small};
  }
`;
