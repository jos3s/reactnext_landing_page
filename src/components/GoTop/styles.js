import styled from 'styled-components';

export const Container = styled.a`
  position: fixed;
  background: ${({ theme }) => theme.colors.primaryColor};
  color: ${({ theme }) => theme.colors.white};
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
`;
