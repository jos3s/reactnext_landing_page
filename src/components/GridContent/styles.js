import styled from 'styled-components';

export const Container = styled.div`
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
`;

export const Html = styled.div`
  margin: ${({ theme }) => theme.spacings.large} 0;
`;
