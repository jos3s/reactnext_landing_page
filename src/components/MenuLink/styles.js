import styled from 'styled-components';

export const Container = styled.a`
  color: ${({ theme }) => theme.colors.primaryColor};
  font-size: ${({ theme }) => theme.font.sizes.small};
  padding: ${({ theme }) => theme.spacings.small};
  text-decoration: none;
  display: block;
  position: relative;

  &::after {
    content: '';
    position: absolute;
    bottom: 0.76rem;
    width: 0;
    left: 50%;
    height: 0.2rem;
    background: ${({ theme }) => theme.colors.secondaryColor};
    transition: all 300ms ease-in-out;
  }

  &:hover::after {
    width: 50%;
    left: 25%;
  }
`;
