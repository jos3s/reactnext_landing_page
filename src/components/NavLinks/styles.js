import styled from 'styled-components';

export const Container = styled.div`
  background: #fff;
  display: flex;
  flex-flow: row wrap;

  @media ${({ theme }) => theme.media.lteMedium} {
    flex-flow: column wrap;
    align-content: center;
  }
`;
