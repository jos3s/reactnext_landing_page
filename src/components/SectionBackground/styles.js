import styled from 'styled-components';

export const Container = styled.div`
  background: ${({ theme, backgroundDark }) =>
    backgroundDark ? theme.colors.primaryColor : theme.colors.white};
  color: ${({ theme, backgroundDark }) =>
    backgroundDark ? theme.colors.white : theme.colors.primaryColor};

  min-height: 100vh;
  display: flex;
  align-items: center;
`;
