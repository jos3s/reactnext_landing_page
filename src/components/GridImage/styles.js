import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../Heading/styles';

export const Container = styled.div`
  ${TextComponent} {
    margin-bottom: ${({ theme }) => theme.spacings.xhuge};
  }
`;

export const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: ${({ theme }) => theme.spacings.large};
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms easy-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
