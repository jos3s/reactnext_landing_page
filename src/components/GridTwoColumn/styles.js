import styled from 'styled-components';

import { Title } from './../Heading/styles';

export const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr 2fr;
  align-items: center;
  gap: ${({ theme }) => theme.spacings.large};

  @media ${({ theme }) => theme.media.lteMedium} {
    grid-template-columns: 1fr;
    text-align: center;
  }

  ${Title} {
    margin-bottom: ${({ theme }) => theme.spacings.xlarge};
  }
`;

export const TextContainer = styled.div`
  @media ${({ theme }) => theme.media.lteMedium} {
    margin-bottom: ${({ theme }) => theme.spacings.large};
  }
`;

export const ImageContainer = styled.div``;

export const Image = styled.img`
  width: 100%;
`;
