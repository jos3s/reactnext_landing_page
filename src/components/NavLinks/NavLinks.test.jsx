import { screen } from '@testing-library/react';
import { NavLinks } from '.';
import { renderTheme } from '../../styles/renderTheme';
import { theme } from '../../styles/theme';

import mock from './mock.js';

describe('<NavLinks />', () => {
  it('should render links', () => {
    renderTheme(<NavLinks links={mock}></NavLinks>);
    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });
  it('should not render links', () => {
    renderTheme(<NavLinks />);
    expect(screen.queryAllByText(/links/i)).toHaveLength(0);
  });
  it('should render links', () => {
    renderTheme(<NavLinks links={mock}></NavLinks>);
    expect(screen.getByText(/link 10/i).parent).toHaveStyleRule('flex-flow', 'column wrap', {
      media: theme.media.lteMedium,
    });
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={mock}></NavLinks>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
