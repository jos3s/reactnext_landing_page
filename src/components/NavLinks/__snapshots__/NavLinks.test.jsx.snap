// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c1 {
  color: #0A1128;
  font-size: 1.6rem;
  padding: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.76rem;
  width: 0;
  left: 50%;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="#target1"
    target="_self"
  >
    link 1
  </a>
  <a
    class="c1"
    href="#target2"
    target="_self"
  >
    link 2
  </a>
  <a
    class="c1"
    href="#target3"
    target="_self"
  >
    link 3
  </a>
  <a
    class="c1"
    href="#target4"
    target="_self"
  >
    link 4
  </a>
  <a
    class="c1"
    href="#target5"
    target="_self"
  >
    link 5
  </a>
  <a
    class="c1"
    href="#target6"
    target="_self"
  >
    link 6
  </a>
  <a
    class="c1"
    href="#target7"
    target="_self"
  >
    link 7
  </a>
  <a
    class="c1"
    href="#target8"
    target="_self"
  >
    link 8
  </a>
  <a
    class="c1"
    href="#target9"
    target="_self"
  >
    link 9
  </a>
  <a
    class="c1"
    href="#target10"
    target="_self"
  >
    link 10
  </a>
</nav>
`;
